### PR TITLE
Move HTTP mock setup to BasicTest class

### PR DESCRIPTION
**Pull request by joseconsador**
Move HTTP mock setup to BasicTest class
/cc @miogalang @atroche 

### References
 - Jira link: 

### Risks
 - None

### CODE REVIEW

File: tests/Zendesk/API/LiveTests/BasicTest.php [hunk 1]
```diff
@@ -34,6 +34,39 @@ public function __construct()
...
+                ->pathIs('/api/v2' . $path)
+            ->then()
+                ->body(json_encode($response))
+                ->statusCode($code)
+            ->end();
+        $this->http->setUp();
+    }
+
+    public function setUp()
+    {
```
Comment (miogalang) [inline tests/Zendesk/API/LiveTests/BasicTest.php]: add parent::setUp and parent::tearDown to be safe

Comment (joseconsador) [inline tests/Zendesk/API/LiveTests/BasicTest.php]: :haircut:

**Comment by miogalang** [comment]
:+1:

**Comment by atroche** [comment]
:+1:

**Pull request merged**